jujugui/static/gui/src/app/components/entity-files/entity-files.js

```diff
@@ -64,17 +118,53 @@ YUI.add('entity-files', function() {
       @return {Array} The markup for the linked files.
     */
     _generateFileItems: function(files, url) {
-      var fileItems = files.map(function(file) {
-        var fileLink = `${url}/${file}`;
-        return (
-          <li key={file} className="entity-files__file section__list-item">
-            <a href={fileLink} target="_blank">
-              {file}
-            </a>
-          </li>
-        );
+      var filetree = this._buildFiletree(files);
+      var clickHandler = this._onDirectoryClick.bind(this);
+
+      /**
+        Recursively create a nested list.
+
+        @method buildList
+        @param {String} path The current path.
+        @param {Array} children All child files and directories located under
+        the current path.
+        @return {Array} The nested markup for the list of files.
+      */
+      function buildList(path, children) {
+        var fileName = path.split('/').pop();
+        if (children === null) {
+          var fileLink = `${url}/${path}`;
+          return (
+            <li key={path} className="entity-files__file">
+              <a href={fileLink} target="_blank">
+                {fileName}
+              </a>
+            </li>
+          );
+        } else {
+          var childItems = [];
+          Object.keys(children).forEach(function(child) {
+            childItems.push(buildList(`${path}/${child}`, children[child]));
+          });
+          return (
+            <li key={path}
+              className="entity-files__directory collapsed"
+              onClick={clickHandler}>
```

Comment:
Clickable items need `tabIndex="0" role="button"` for keyboard navigation and accessibility.